### ISSUE TITLE
Changing a player's position on the player list

### ISSUE THREAD

**Issue by FrescoPL** [haxball/haxball-issues#390]
Hey, I have an idea that would help me on a headless room I'm developing.

I created an anti-afk system that moves a player to Spectators after they write !afk on the chat. However, when this happens between matches (during the picking phase), they land on the bottom of the Spectators list, which in turn breaks the picking queue on the room.

I could return them to their previous position "manually", by moving the players above them to a team and back to Spectators, but it would create spam on the chat. This can be fixed in a few ways.
First of all, you can add an option to temporarily disable chat notifications. This, when used for my purpose, would load my room more than necessary, but it may be useful in many other situations.
A second option is adding a possibility to move players within the same team in addition to moving them between teams.
Last but not least, you could create an option to make a player unable to be moved between teams by Administrators.

I hope that you'll implement at least one of the things I mentioned.

With love,
Fresco

**Comment by cr0ndor** [comment]
Leo,

In every room I've ever played in, the specs list is used to show the order that the players entered, and thus how long they have been waiting. The winning team stays (or is moved to red), while the person at the top of the specs list will get to play and will pick their team.

If, for example, you move the wrong person, when you correct the error they will be at the bottom unless you manually move everyone else out of specs and then back, or just try to remember that the person at the bottom has been waiting the longest, etc.

It's just a quality-of-life suggestion.

**Comment by olegmisar** [comment]
Agree with @cr0ndor. @basro, you can make something like this: `room.setPlayersOrder(playerIDs: int[])`. So, we will be able to do this:
```javascript
const players = room.getPlayerList()
players.sort(sortingFunction)
const playerIDs = players.map(p => p.id)
room.setPlayersOrder(playerIDs)
```

**Comment by basro** [comment]
I agree this is a useful thing and I'll probably implement some api for changing the order of players.
I'm not sure what the api will be like at the moment.

**Comment by Leo-MapCreator** [comment]
I didnt understand that when he said, cr0ndor, for me it had nothing to do with what you said, if the person was afk, it wouldnt change anything if the person would be dragged to the bottom

**Comment by Leo-MapCreator** [comment]
@basro I have and idea for that using API, when you work with API, you work with almost instantly commands, one of the commands I had was to put host to spec if his team wasnt spec, and if you tried to move him to another team, we would instantly be dragged to the bottom of the spec, you can't even see him in one of the teams, so with that in mind, what you can do is remove everyone from spec, recording their current position, then organize the players in order, putting the player you want in the position you want, but you gotta remove the indicator that a player was moved to 'team X' otherwise it will spam the chat

**Comment by evi1a** [comment]
Has been almost half a year since the feature request got approved
